integrationtests/tests/fixtures/mod.rs

```diff
@@ -232,70 +227,85 @@ pub trait LightningTest {
-                .expect("Could not parse URL to generate GatewayClientConfig API endpoint"),
-        };
+        let ln_config = rpc_config_mx
+            .lock()
+            .await
+            .config()
+            .expect("Failed to get rpc config");
 
-        let database: Database = MemDatabase::new().into();
+        // Build user client and test
         let user_cfg = UserClientConfig(client_config.clone());
         let user_client = UserClient::new(user_cfg.clone(), database.clone(), Default::default());
-        let user = UserTest {
+        let user_test = UserTest {
             client: user_client,
             config: user_cfg,
         };
 
-        let bind_addr: SocketAddr = format!("127.0.0.1:{}", bind_port).parse().unwrap();
-        let gw_cfg = GatewayClientConfig {
-            client_config: client_config.clone(),
+        // Define keys
+        let keys = LightningGateway {
+            mint_pub_key: kp.public_key(),
+            node_pub_key: ln_config.pub_key,
+            api: Url::parse("http://example.com")
+                .expect("Could not parse URL to generate GatewayClientConfig API endpoint"),
+        };
+
+        // Build gateway client
+        let gw_client_cfg = GatewayClientConfig {
+            client_config,
             redeem_key: kp,
             timelock_delta: 10,
-            api: Url::parse(format!("http://{}", bind_addr).as_str())
+            api: Url::parse(format!("http://{}", ln_config.bind_addr).as_str())
                 .expect("Could not parse URL to generate GatewayClientConfig API endpoint"),
-            node_pub_key,
+            node_pub_key: ln_config.pub_key,
         };
-        let client = Arc::new(GatewayClient::new(
-            gw_cfg,
+
+        let gateway_client = Arc::new(GatewayClient::new(
+            gw_client_cfg,
             database.clone(),
             Default::default(),
         ));
-        let (sender, receiver) = tokio::sync::mpsc::channel::<GatewayRequest>(100);
-        let adapter = Arc::new(ln_client_adapter);
-        let ln_client = Arc::clone(&adapter);
-        let gateway = LnGateway::new(client.clone(), ln_client, sender, receiver, bind_addr);
-        // Normally, this client registration with the federation is automated as part of running the gateway
-        // In test cases, we want to register without running a gateway
-        client
-            .register_with_federation(client.config().into())
+
+        // Register federation and keep reference to a gateway actor
+        let actor = gateway
+            .register_federation(gateway_client.clone())
             .await
-            .expect("Failed to register client with federation");
+            .expect("Failed to register federation");
+
+        // Run gateway : NOTE: We never run gateways in tests because it starts a blocking loop
+        //
+        // gateway.run().await.expect("gateway failed to run");
+
+        // Provide an adapter to the gateway ln rpc
```

Comment:
@NicolaLS, I just realized I have not implemented the adapter correctly. Calling `adapter.fail_invoice()` will have no effect on gateway test's ln rpc since the adapting happens out here on an ln_rpc feference that is not used by the gateway test... will fix